primary/src/block_synchronizer/responses.rs

```diff
@@ -109,7 +109,17 @@ impl<PublicKey: VerifyingKey> CertificatesResponse<PublicKey> {
         let invalid_certificates: Vec<Certificate<PublicKey>> = peer_found_certs
             .clone()
             .into_iter()
-            .filter(|c| c.verify(committee).is_err())
+            .filter(|c| {
+                if let Err(err) = c.verify(committee) {
+                    error!(
+                        "Certificate verification failed for id {} with error {:?}",
+                        c.digest(),
+                        err
+                    );
+                    return true;
+                }
+                false
+            })
```

Comment:
That should be handled more properly rather than just a log message. Added this issue here: https://github.com/MystenLabs/narwhal/issues/260